app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,20 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find_by_id(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+    @comment.author = current_user
+
+    if @comment.save
+      flash[:success] = 'Comment added!'
+      redirect_to user_post_path(current_user, @comment.post)
+    else
+      flash[:errors] = @comment.errors.full_messages
+    end
+  end
+
+  private
```

Comment:
- Hi, Nice job in creating a comment on behalf of the currently logged-in user (current_user) of the app. I would advice you to add `new` actions like what you did with `create` action.
**Hint:**
 ```ruby
def new
    @comment = Comment.new
  end
```
- Also, create a form in a directory such as `views/comments/new.html.erb`  which accepts input for creating comments.

Reply:
I don't understand. Why do I need new action just to create new comments? I presume that we don't need a new directory to create comments. Could you please kindly explain why we need new action and its purpose? @OLIPLICHE

Reply:
- [ ] As required by the last CR please define the new method for the comments. Why does it use 

```ruby
def new 
  @comment = Comment.new 
end
```

?

The def new method in the comments controller (CommentsController) is used to prepare and display a form for users to create new comments. Within this method, a new instance of the Comment class is created and assigned to the instance variable @comment. This variable will be used later in the comment creation form in the corresponding view.

The purpose of this action is that when users access the comment creation page (e.g., by clicking an "Add Comment" button in the view of a post), the controller prepares a new empty instance of Comment so that the user can write their comment in the form.

Reply:
Thank you @franclobo